src/libexpr/eval.cc

```diff
@@ -1666,18 +1667,19 @@ void EvalState::callFunction(Value & fun, size_t nrArgs, Value * * args, Value &
                 for (size_t i = 0; i < argsLeft; ++i)
                     vArgs[argsDone + i] = args[i];
 
-                auto name = primOp->primOp->name;
+                auto fn = primOp->primOp;
                 nrPrimOpCalls++;
-                if (countCalls) primOpCalls[name]++;
+                if (countCalls && !fn->hideInDiagnostics) primOpCalls[fn->name]++;
```

Comment:
Technically a primop call, so that should be ok.

```suggestion
                // This will count black holes, but that's ok, because unrecoverable errors are rare.
                if (countCalls) primOpCalls[fn->name]++;
```